spec/rubocop/cop/lint/useless_assignment_spec.rb

```diff
@@ -266,6 +266,16 @@ def some_method
         end
       RUBY
     end
+
+    it 'registers offenses for the assignment in numblock' do
+      expect_offense(<<~'RUBY')
+        1.times do
+          puts "Iteration #{_1 + 1}"
+          foo = 2
+          ^^^ Useless assignment to variable - `foo`.
+        end
+      RUBY
```

Comment:
Is it possible to reproduce the error with this test case? I reverted `:numblock` from the `case` / `when` branch in the scope.rb and ran the test, but the error was not reproduced.

Reply:
It's still hard for me to get the exact repro I get on my codebase. Let me try a bit more.

Reply:
```ruby
def validate_format(foo)
  return false unless foo.in_format?

  bar = [1]
  baz = 1
  bap = 2

  baq = 0
  (0..10).each { baq += bar[_1] * foo[_1].to_i }

  baq % baz == bap
end
```

This is the code that crashes because of the `numblock` locally, but I still cannot reproduce it in isolation. @koic I'm posting the code if you have an idea here.